File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    r=[]
    cnt,zcnt=0,0
    while int(s,2)!=1:
        cnt+=1
        zcnt+=s.count('0')
        s=len(str(s).replace('0',''))
        s=bin(s)[2:]
    r=[cnt,zcnt]
    return r

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>
#include <cmath>
#include <algorithm>

using namespace std;

vector<int> solution(string s) {
    vector<int> r;
    int cnt=0,zcnt=0,n=0;
    while(s.size()>1) {
        cnt+=1;
        zcnt+=count(s.begin(),s.end(),'0');
        while(s.find('0')!=string::npos) {
            s.replace(s.find("0"),1,"");
        }
        n=s.size();
        s="";
        while(n>0) {
            s+=to_string(n%2);
            n/=2;
        }
    }
    r={cnt,zcnt};
    return r;
}